WIP Closes #1751

djcelaya opened the pull request.

mmunro-ltrr reviewed (dismissed):
The only difference from the mockup seemed to be the alignment of the secondary items in the dropdown. Here, they're aligned with the primary item text, but in the mockup they're indented a bit relative to it. The tertiary item alignment seemed similar in both.

bberndt-uaz reviewed (commented):
This looks great! The scss is very clean and organized.

1. In the mockup, the top-level dropdowns open on hover instead of on click. Is that a requirement?
2. I recommend adding a second, simplified example showing the styling of the navbar when the active page is one of the children in the dropdown (like on [this mockup page](https://xd.adobe.com/view/b5dea83d-f569-4f51-9a42-c5b75ea82ece-e08d/screen/879ebdca-dd20-4777-8cdc-4b003be056df?fullscreen&hints=off)).
3. Another use case to consider (though I don't think we need an example of it) is when a top menu item is not a page (like in our demo content). In that case, I believe both the text and the dropdown button are supposed to open the dropdown.
4. What do you think about the red underlines on the top menu items? I know it was part of the requirements, but I still think it seems sort of inconsistent with the blue underlines on the tertiary links in the dropdowns (and our other links).

danahertzberg commented:
@djcelaya LOVELY!!!! This is looking fantastic. All of my previous comments have been sufficiently accomplished. :) 

Here are some new thoughts lol.

1. On the active tertiary style, please add an underline.
2. On the active tertiary style, please remove the bold, but keep the black color.
3. When a secondary nav element is active, is it possible to have the tertiary menu expanded by default when the primary nav is opened?
   - See [mockup](https://xd.adobe.com/view/b5dea83d-f569-4f51-9a42-c5b75ea82ece-e08d/screen/99a9810d-5e68-459d-95d0-246d0193d022?fullscreen&hints=off) > Programs > MBA Programs 
4. Similar to 3. When a tertiary nav element is active, is it possible to have the tertiary menu expanded by default when the primary nav is opened?
   - See [mockup](https://xd.adobe.com/view/b5dea83d-f569-4f51-9a42-c5b75ea82ece-e08d/screen/99a9810d-5e68-459d-95d0-246d0193d022?fullscreen&hints=off) > Programs > MBA Programs > Full-time MBA - Tucson
5. Super low priority. I noticed the "click" state is the same as active. Is that normal? I would think it should match the hover state instead. 🤔

djcelaya commented:
> @djcelaya LOVELY!!!! This is looking fantastic. All of my previous comments have been sufficiently accomplished. :)
> 
> Here are some new thoughts lol.
> 
> 1. On the active tertiary style, please add an underline.
> 2. On the active tertiary style, please remove the bold, but keep the black color.
> 3. When a secondary nav element is active, is it possible to have the tertiary menu expanded by default when the primary nav is opened?
>    
>    * See [mockup](https://xd.adobe.com/view/b5dea83d-f569-4f51-9a42-c5b75ea82ece-e08d/screen/99a9810d-5e68-459d-95d0-246d0193d022?fullscreen&hints=off) > Programs > MBA Programs
> 4. Similar to 3. When a tertiary nav element is active, is it possible to have the tertiary menu expanded by default when the primary nav is opened?
>    
>    * See [mockup](https://xd.adobe.com/view/b5dea83d-f569-4f51-9a42-c5b75ea82ece-e08d/screen/99a9810d-5e68-459d-95d0-246d0193d022?fullscreen&hints=off) > Programs > MBA Programs > Full-time MBA - Tucson
> 5. Super low priority. I noticed the "click" state is the same as active. Is that normal? I would think it should match the hover state instead. 🤔

1. Done.
2. Done.
3. I don't think this is possible with CSS alone. I adjusted the markup on the second and third level demo pages so the respective submenus are expanded (added `.show` class and adjusted `aria-expanded` label). I think this logic (dynamically changing the HTML) could be handled by Twig on the Drupal side of things. Or, some custom JavaScript may do it as well. At a glance, it looks like we may be handling this with Twig in our [current implementation](https://github.com/az-digital/az_quickstart/blob/main/themes/custom/az_barrio/templates/navigation/menu--main.html.twig#L59). What do you think?
4. See 3.
5. TBD

djcelaya commented:
> A couple potential follow-up issues, unless you want to address one or both of them sooner!
> 
> <img alt="image" width="406" height="434" src="https://private-user-images.githubusercontent.com/74572157/523962005-5f992042-2f16-4765-b523-5332115b04cc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUzMjEyNzAsIm5iZiI6MTc2NTMyMDk3MCwicGF0aCI6Ii83NDU3MjE1Ny81MjM5NjIwMDUtNWY5OTIwNDItMmYxNi00NzY1LWI1MjMtNTMzMjExNWIwNGNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjA5VDIyNTYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjNzUxOTJjYWY3YjFkMDE1NzBmMjEyNGI2ZjljOWRjNzc2OTcxNDQzOWU0NDNlNmJjZTI2MGU1NzE1MGQxZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CZZdV05K2WP3Ryi6KY-cXDrQRr4La8G8iGRd7GsqnTo">
> 1. I see that the mockup uses sharp corners for the dropdowns and that Dana requested rounded corners in the comments on this PR. This results in the top border of the dropdown curving at the ends. I think it would be best to address this before the Quickstart 3.2.0 stable release.
> 2. It looks like hover changes the link color to blue on all links except for tertiary links, where the text remains azurite and the underline is blue.

1. During a meeting with central marketing on Dec 9, we decided that the curving was visually acceptable for now.
2. This will be addressed in #1834. Thanks!
